src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -363,14 +356,27 @@ class CDCL(val clauses: MutableList<MutableList<Int>>, initNumber: Int = 0) {
                     }
                 }
             }
-            addVariable(clause, lit)
+            setVariableValues(clause, lit)
         }
 
         return -1
     }
 
+    // add a variable to the trail and update watchers of clauses linked to this variable
+    private fun setVariableValues(clause: Int, lit: Int): Boolean {
+        if (getStatus(lit) != VarStatus.UNDEFINED) return false
```

Comment:
You should not return false in the case, when `lit` is already assigned `true`, since it does not contradict the intention of this method: you wanted to assign lit to true — it is already true — so just return okay. The only "bad" situation (when you would `return false`) is when your assignment (via this method, which is implicitly `lit = true`) contradicts the current one (`getStatus(lit)==VarStatus.False`).